mautic/mautic
# Add Contact ID column into form results table

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | /
| Deprecations? | /

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
The submission date value used to contain a link to the contact who made the submission. This link stopped working after https://github.com/mautic/mautic/pull/5792 was merged in 2.13.1.

This PR adds the link back but as a new `Contact ID` column. It was confusing that the link to the contact is on the submission date. I also saw a discussion where some community member thought that the ID column displays Contact IDs, but it displays submission IDs. I renamed the column label of `ID` to `Submission ID` so this confusion won't happen again.

This PR also escapes values from form submissions.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Go to a form result page where you have some submissions. If you don't have such form, create one and submit it in an incognito browser several times.
2. Notice that there is no link which leads to the contact who submitted the results.

#### Steps to test this PR:
1. Refresh the page
2. The new `Contact ID` column is here and it contains IDs with links to the contacts.
3. Make sure the new column is searchable and orderable and the links are working.
